packages/magner/lib/types/form/base.ts

```diff
@@ -27,17 +27,17 @@ export interface BaseProps {
 
   /** Whether the field is disabled. Accepts static boolean or the dynamic
    * MixedChecker function with access to the user's role and form's state */
-  disabled?: boolean | MixedChecker,
+  disabled?: MixedChecker,
   /** Pure disabled function */
   disabledCondition?: MixedChecker,
```

Comment:
Suggestion: create an interface that would extend the `BaseProps` (with *Condition props), but be used for the inner purposes. In this case, users won't see these properties, but they will exist in the inner context.

For example, imagine it like so:

```ts
export interface BasePropsInner extends BaseProps {
  disabledCondition?: MixedChecker,
  requiredCondition?: MixedChecker,
  readonlyCondition?: MixedChecker,
}
```

And why did you remove the `boolean` flag?

Reply:
@VanishMax I removed the boolean flag since the MixedChecker function returns a boolean value anyway.